ci/changelog_helper.py

```diff
@@ -3,16 +3,32 @@
 import subprocess
 import sys
 
+USAGE = """
```

Comment:
Can we just use the GitHub-generated release notes instead?

If not, this should get rewritten into Rust...

Reply:
Of course I've thought about using the GitHub-generated changelog, but we still need to update the Rust blog so looks like we require this kind of manual "polishing" process somewhere in our workflow anyway.

I think we can afford to rewrite it in Rust. This script has already reached a point where I think I have a basic idea about what the Rust version should look like. However, I'm concerned about two things:

- Startup time: I think this is the main reason for which the existing scripts in this repo are written in Bash, PowerShell, Python and Perl.
- Execution method: Currently `cargo-script` isn't there yet. Is there a nice way to do a single-file Rust script that you prefer? Also, do we have to stick to the standard library with this method?

Do you have a concrete example in mind that I can refer to?

Reply:
We should stick it in `examples`, and use `dev-dependencies` to add extra dependencies for it (like octocrab).

I don't think we should necessarily make the blog posts on the Rust blog actually like they have been before. In particular, IMO listing all contributors in the blog post is overkill (the Rust blog doesn't do that for Rust releases) and could be deferred to the changelog here.

In most of the other crates I maintain we use GitHub releases instead of a `ChangeLog.md` file, which is pretty nice and integrates the generated changelogs.

None of this should block 1.27.1, let's (finally) get that thing released!

Reply:
> I don't think we should necessarily make the blog posts on the Rust blog actually like they have been before. In particular, IMO listing all contributors in the blog post is overkill (the Rust blog doesn't do that for Rust releases) and could be deferred to the changelog here.

@djc Yeah. It's kinda tedious really. I wish I wouldn't have to spend like an hour per release night just to figure this out. Maybe the release blog posts are written in this way because... People tend to do what the Romans do to be safe, but things don't have to be this way.

> In most of the other crates I maintain we use GitHub releases instead of a `ChangeLog.md` file, which is pretty nice and integrates the generated changelogs.

I do this with my projects as well!

But at this point I still think Rustup is not "my project", and things are where they are... for a (probably historical) reason. Given that, I think the best thing I can do in this respect is migrating Rustup to [Release PRs](https://github.com/googleapis/release-please?tab=readme-ov-file#whats-a-release-pr) somewhere down the line, which allows auto-maintenance of `CHANGELOG.md`.